src/sentry/models/rulesnooze.py

```diff
@@ -0,0 +1,24 @@
+from django.db import models
+
+from sentry.db.models import FlexibleForeignKey, Model, region_silo_only_model, sane_repr
+
+
+@region_silo_only_model
+class RuleSnooze(Model):
+    """
+    Whether or not an issue alert or metric alert is snoozed for a user and optionally for how long.
+    """
+
+    __include_in_export__ = True
+
+    user = FlexibleForeignKey("sentry.User")
+    rule = FlexibleForeignKey("sentry.Rule", null=True)
+    alert_rule = FlexibleForeignKey("sentry.AlertRule", null=True)
+    muted = models.BooleanField(default=True)
+    until = models.DateTimeField(null=True)
```

Comment:
Is there a "forever" datetime? Or we could consider a null `until` to be forever? That way we wouldn't need both columns.

Reply:
Yeah, let's go with `null` to be forever, that's consistent with group snooze